[#143] view: 진열장 화면 구현

gustn3965 opened the pull request:
## 개요
진열장 화면 구현 

## 시연  ( 리뷰 반영한 후  ) 
![drawer](https://user-images.githubusercontent.com/48749182/139396724-d64d2112-6e7c-4947-922a-9614dbe4e390.gif)

## 작업 사항

- `DrawerViewController` 구현 ( 진열장 진입 화면 뷰컨트롤러 )
- `SelectingViewController` 구현 (진열장 특정 아이템 보여주는 뷰컨트롤러)
- `DrawerCoordinator` 구현 ( 진열장 Navigation )
- `ImageWithTwoLabellVerticalCetnerXView` 구현 ( 진열장 아이템 뷰 )
    -  이 뷰를 이용하여 재사용했습니다. 
    - `DrawerViewController` 상단과, 하단의 컬렉션뷰의 cell로 사용됐고, `SelectingViewController`에서도 사용됩니다. 
- 추가적으로 병연님의 피드백으로, `DrawerViewController`의 상단 대표 화면을 스크롤되게 하기 위해, 이를 `DrawerHeaderView` 로 만들어서 구현했습니다. 

## 특이사항 
- `DrawerViewController`내의 컬렉션뷰의 cell의 뷰는 test로 랜덤하게 매번 구성요소가 바뀝니다. 
- 마찬가지로 `SelectingViewController`내의 뷰 구성요소도 랜덤하게 매번 바뀝니다. 
## 예외사항 
- 단순히 화면 구현 및 화면 이동 로직만 구현했습니다. 


### Linked Issue

close #143 
close #150

gustn3965 committed:
[#143] view: 진열장 화면 구현
- DrawerViewController 구현 ( 진열장 진입 화면 뷰컨트롤러 )
- SelectingViewController 구현 (진열장 특정 아이템 보여주는 뷰컨트롤러)
- DrawerCoordinator 구현 ( 진열장 Navigation )
- ImageWithTwoLabellVerticalCetnerXView 구현 ( 진열장 아이템 뷰 )

jwonyLee commented on ThingLog/ViewController/Drawer/DrawerViewController.swift:
```diff
@@ -0,0 +1,147 @@
+        let editButton: UIButton = UIButton()
+        editButton.setTitle("확인", for: .normal)
+        editButton.titleLabel?.font = UIFont.Pretendard.body1
+        editButton.setTitleColor(SwiftGenColors.black.color, for: .normal)
```
피그마에서는 "확인"이 아니라 "완료"이고, 완료 버튼 폰트는 `title1` 이에요~

jwonyLee commented on ThingLog/ViewController/Drawer/DrawerViewController.swift:
```diff
@@ -0,0 +1,147 @@
+        
+        let backButton: UIButton = UIButton()
+        backButton.setImage(SwiftGenAssets.paddingBack.image, for: .normal)
+        backButton.tintColor = SwiftGenColors.black.color
```
피그마에서는 X 이미지를 사용하고 있던데 한 번 확인해주세요. :eyes:

jwonyLee commented on ThingLog/ViewController/HomeViewController.swift:
```diff
@@ -209,6 +210,16 @@ extension HomeViewController {
         .disposed(by: disposeBag)
     }
     
+    /// 진열장 화면으로 이동하는 메서드다.
+    func subscribeDrawerImageView() {
```
탭 바가 하단에 보여지고 있는데 화면 설계서를 보면 설정 화면 처럼 탭 바를 숨겨야 할 거 같아요~

jwonyLee reviewed (approved):
LGTM :+1:
간단하게 코멘트 남겼습니당~ 수고하셨습니다~~

gustn3965 committed:
[#143] ref: 지원님, 병연님 피드백 반영
- 탭바 우측 버튼 폰트 및 이름 변경
- 탭바 좌측 버튼 이미지 변경
- 진열장 첫화면 하단 탭 숨김
- 진열장 첫화면 상단 대표 진열장화면도 스크롤되게 변경

gustn3965 commented:
꼼꼼하게 봐주셔서 놓친 부분을 수정할 수 있었습니다. 감사합니다. 

모두 반영했습니다.

추가적으로 병연님의 피드백으로, `DrawerViewController`의 상단 대표 화면을 스크롤되게 하기 위해, 이를 `DrawerHeaderView` 로 만들어서 구현했습니다. 
다시한번 리뷰 부탁드립니다.  풀리퀘 글에도 시연 동영상 다시 반영했습니다.

jwonyLee commented on ThingLog/View/HeaderView/DrawerHeaderView.swift:
```diff
@@ -0,0 +1,57 @@
+    // MARK: - View
+    let drawerView: ImageWithTwoLabellVerticalCetnerXView = {
+        let drawerView: ImageWithTwoLabellVerticalCetnerXView = ImageWithTwoLabellVerticalCetnerXView(imageViewHeight: 100)
+        drawerView.translatesAutoresizingMaskIntoConstraints = false
```
지금 다시 보니까 `ImageWithTwoLabelVerticalCenterXView` 에 오타가 있네요. `TwoLabell`

jwonyLee reviewed (approved):
LGTM :+1:

gustn3965 committed:
[#143] ref: 지원님 피드백 반영
- ImageWithTwoLabelVerticalCetnerXView 객체 이름 변경
